codex/marketplace.md

```diff
@@ -0,0 +1,157 @@
+
+`reward`
+
+- SHOULD be a token known to the network.
+- it MUST be paid directly to storage nodes that agree to storage contract
+
+`collateral`
+
+All storage nodes MUST provide token collateral before being able to fulfill a storage contract.
+The collateral will be taken to punish a storage node when it does not follow the contract.
+The following conditions MUST be fulfilled in a Codex storage contract
+- failing to provide proof of storage periodically, the RECOMMENDED method for proofs is [Proof-of-Data-Possession](https://hackmd.io/2uRBltuIT7yX0CyczJevYg?view).
+- a portion of `collateral` MUST be offered as a reward to validator nodes,
+and a portion SHOULD be offered as a reward to storage nodes that repair [slots](#slots).
+
+`proofProbability`
+
+Storage nodes are REQUIRED to provide proofs of storage that are verified by validator nodes.
+The requester SHOULD provide the value for the frequency of proofs provided by storage nodes.
+
+`duration`
+
+- Once the `reward` has depleted from periodic storage node payment,
+the storage request SHOULD end.
+The requester MAY renew the storage request by creating a new request with the same `cid` value.
+- Data MAY be considered lost during contract `duration` if storage nodes fail to provide storage proofs and
+no other storage nodes decide to fill empty slots, see [slots](#slots) below.
+
+`dispersal`
+
+- Storage nodes MUST not allowed to fill all slots in a request.
+- Requester SHOULD provide dispersal value
+
+### Slots
+
+After a storage request is made, 
+the requester's Codex node will create slots using erasure coding.
+Each slot will contain a data block being stored.
+- Storage Nodes MUST provide token collateral and proof of storage to fill a slot
+- If storage nodes fail to provide proof of storage in a given period,
+the slot will become empty and the host assigned to that slot MUST forfeit its `collateral`.
+Other storage nodes can earn the forfeited `collateral` by providing new proof of storage and `collateral`,
+this is referred to as repairing the empty slot.
+
+-----------
+
+        proof &                                 proof &
+      collateral   proof          missed      collateral               missed
+            |        |              |               |                    |
+            v        v              v               v                    v
+            -------------------------------------------------------------------
+     slot:  |///////////////////////|               |////////////////////|
+            -------------------------------------------------------------------
+                                    |                                    |
+                                    v                                    v
+                                collateral                           collateral
+                                  lost                                 lost
+
+
+
+            ---------------- time ---------------->
```

Comment:
The diagram above may need to be adjusted. The way it works:

A storage provider can be miss a certain number of proofs (called [`slashCriterion`](https://github.com/codex-storage/codex-contracts-eth/blob/4d9320a58208f9d9609f6edc3cccdb29619e31ae/contracts/Configuration.sol#L18)) before it is slashed. Once the storage provider is slashed a certain number of times (called [`maxNumberOfSlashes`](https://github.com/codex-storage/codex-contracts-eth/blob/4d9320a58208f9d9609f6edc3cccdb29619e31ae/contracts/Configuration.sol#L17C9-L17C27), the slot is freed and the storage provider's collateral is lost.